posydon/unit_tests/utils/test_gridutils.py

```diff
@@ -0,0 +1,760 @@
+            for j in range(2):
+                for i,v in enumerate(MESA_SH_data[j]):
+                    if v==int(v):
+                        v = int(v)
+                    if i==0:
+                        test_file.write("\n{:>12}".format(v))
+                    else:
+                        test_file.write(" {:>12}".format(v))
+        return
+
+    @fixture
+    def out_path(self, tmp_path):
+        # a temporary out path for testing
+        return totest.os.path.join(tmp_path, "out.txt")
+
+    @fixture
+    def out_file(self, out_path):
+        # a temporary out file for testing
+        with open(out_path, "w") as test_file:
+            for i in range(5):
+                test_file.write("Test HEADER{}\n".format(i+1))
+        return
+
+    @fixture
+    def out_gz_path(self, tmp_path):
+        # a temporary out path for testing
+        return totest.os.path.join(tmp_path, "out_gz.txt")
+
+    @fixture
+    def out_gz_file(self, out_gz_path):
+        # a temporary out file for testing
+        with open(out_gz_path, "w") as test_file:
+            for i in range(5):
+                test_file.write("Test HEADER{}\n".format(i+1))
+            test_file.write("model is overflowing at ZAMS\n")
+        totest.os.system(f"gzip -1 {out_gz_path}")
+        return
+
+    @fixture
+    def out_path2(self, tmp_path):
+        # a temporary out path for testing
+        return totest.os.path.join(tmp_path, "out2.txt")
+
+    @fixture
+    def out_file2(self, out_path2):
+        # a temporary out file for testing
+        with open(out_path2, "w") as test_file:
+            for i in range(5):
+                test_file.write("Test HEADER{}\n".format(i+1))
+            test_file.write("TURNING ON CE\n")
+        return
+
+    @fixture
+    def out_path3(self, tmp_path):
+        # a temporary out path for testing
+        return totest.os.path.join(tmp_path, "out3.txt")
+
+    @fixture
+    def out_file3(self, out_path3):
+        # a temporary out file for testing
+        with open(out_path3, "w") as test_file:
+            for i in range(5):
+                test_file.write("Test HEADER{}\n".format(i+1))
+            test_file.write("Terminate because accretor (r-rl)/rl > "+\
+                            "accretor_overflow_terminate\n")
+        return
+
+    @fixture
+    def out_path4(self, tmp_path):
+        # a temporary out path for testing
+        return totest.os.path.join(tmp_path, "out4.txt")
+
+    @fixture
+    def out_file4(self, out_path4):
+        # a temporary out file for testing
+        with open(out_path4, "w") as test_file:
+            for i in range(5):
+                test_file.write("Test HEADER{}\n".format(i+1))
+            test_file.write("TURNING ON CE\n")
+            test_file.write("Terminate because accretor (r-rl)/rl > "+\
+                            "accretor_overflow_terminate\n")
+        return
+
+    @fixture
+    def out_path5(self, tmp_path):
```

Comment:
It's quite unclear what fixtures are actually required for each test. 
Because they're grouped together in one class, it's quite unclear. I would suggest splitting up the functions with the same dependencies into different test classes. This should make it much cleaner to see keep an overview.